src/state_machine/coordinator/fire.rs

```diff
@@ -633,28 +654,49 @@ impl<Aggregator: AggregatorTrait> Coordinator<Aggregator> {
-                                    &self.config.signer_public_keys[bad_signer_id];
+                                let signer_public_key =
+                                    compute::point(&self.config.public_keys.signers[signer_id])?;
+
+                                let Some(bad_signer_ecdsa_key) =
+                                    self.config.public_keys.signers.get(bad_signer_id)
+                                else {
+                                    warn!("Signer {signer_id} reported BadPrivateShare from invalid signer_id {bad_signer_id}, mark {signer_id} as malicious");
+                                    self.malicious_dkg_signer_ids.insert(*signer_id);
+                                    continue;
+                                };
+                                let bad_signer_public_key = compute::point(bad_signer_ecdsa_key)?;
                                 let mut is_bad = false;
 
                                 if bad_private_share.tuple_proof.verify(
-                                    signer_public_key,
-                                    bad_signer_public_key,
+                                    &signer_public_key,
+                                    &bad_signer_public_key,
                                     &bad_private_share.shared_key,
                                 ) {
                                     // verify at least one bad private share for one of signer_id's key_ids
                                     let shared_secret =
                                         make_shared_secret_from_key(&bad_private_share.shared_key);
 
-                                    let dkg_public_shares = &self.dkg_public_shares[bad_signer_id]
+                                    let Some(bad_dkg_public_shares) =
+                                        self.dkg_public_shares.get(bad_signer_id)
+                                    else {
+                                        warn!("Signer {signer_id} reported BadPrivateShare from signer {bad_signer_id} who didn't send public shares , mark {signer_id} as malicious");
+                                        self.malicious_dkg_signer_ids.insert(*signer_id);
+                                        continue;
+                                    };
+                                    let dkg_public_shares = &bad_dkg_public_shares
                                         .comms
                                         .iter()
                                         .cloned()
                                         .collect::<HashMap<u32, PolyCommitment>>();
-                                    let dkg_private_shares =
-                                        &self.dkg_private_shares[bad_signer_id];
-                                    let signer_key_ids = &self.config.signer_key_ids[signer_id];
+                                    let Some(dkg_private_shares) =
+                                        &self.dkg_private_shares.get(bad_signer_id)
```

Comment:
I don't think that you need the reference here.

Reply:
Fixed in a3c5c3276cb9336959caf47f9842389d9f9beb39